### DISCUSSION TITLE
Making sure the user knows they're using the PublicAPI

### DISCUSSION THREAD

**Discussion by allerter** [johnwmillr/LyricsGenius#211] [Ideas]
As you know, the PublicAPI (or the Web API as Genius calls it) isn't officially supported by Genius and not only may change without notice, users that send a lot of requests might hit 403 errors. Even IPs that Genius considers to be a bot might get hit a 403 error. That might be okay for some, but the user should be aware that they're using a part of the API that isn't part of the official Developers API.
My suggestion is to add a parameter called `use_public_api` to `Genius.__init__` with the default value of False. Whenever the user tries to use a method that either directly or indirectly uses the Public API, we'll check the value of the attribute to see if we can use the Public API or not.
```python
>>> genius = Genius(TOKEN, use_public_api=False)
>>> genius.search_artist("Eminem")
```
```
Searching for songs by Eminem...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "lyricsgenius\genius.py", line 525, in search_artist
    artist_id = artist_id if artist_id else find_artist_id(artist_name)
  File "lyricsgenius\genius.py", line 510, in find_artist_id
    response = self.search_all(search_term)
  File "lyricsgenius\api\public_methods\search.py", line 210, in search_all
    raise Exception("Can't use Public API method `search_all` unless"
Exception: Can't use Public API method `search_all` unless `genius.use_public_ap
i=True`.
Read more at ...
```

**Comment by johnwmillr** [comment]
I like this suggestion, especially `use_public_api` defaulting to False.

**Comment by allerter** [comment]
Added #212.
